src/drawers/areaDrawer.ts

```diff
@@ -16,7 +16,7 @@ export module Drawers {
     }
 
     public selectionForIndex(index: number) {
-      return this.renderArea().select(this.selector());
```

Comment:
But this was the main source of improvement. If I do what you suggested, I have no more caching. Plus, the `d3.select()` and `whateverD3SelectionHere.select()` are essentially the same thing. By looking at the argument passed (an Element, not a string), you can tell that it will not look throughout the entire page